strr-host-pm-web/app/pages/application.vue

```diff
@@ -263,8 +263,28 @@ const handleSubmit = async () => {
     // if all steps valid, submit form with store function
     if (isApplicationValid) {
       shouldSkipConfirmModal = true
-      const { paymentToken, filingId, applicationStatus } = await submitApplication(false, applicationId.value)
-      const redirectPath = `/dashboard/${filingId}`
+      const {
+        paymentToken,
+        filingId,
+        applicationStatus,
+        registrationId,
+        registrationNumber,
+        applicationType
+      } = await submitApplication(false, applicationId.value)
+
+      // Determine redirect path based on feature flag and application type
+      let redirectPath: string
+      if (isNewDashboardEnabled.value) {
+        if (applicationType === 'renewal' && registrationId && registrationNumber) {
+          permitStore.selectedRegistrationId = String(registrationId)
+          redirectPath = `/dashboard/registration/${registrationNumber}`
```

Comment:
Because of condition `if (isNewDashboardEnabled.value) `, should the path be `/dashboard-new/...`, or middleware suppose to redirect to new dashboard?

Reply:
This has now been addressed with redirect middleware changes.